cli/protoc-gen-grafbase-subgraph/tests/testcases/composite_schemas_entity.proto

```diff
@@ -0,0 +1,104 @@
+//!file:grafbase/options.proto
+syntax = "proto2";
+
+import "google/protobuf/descriptor.proto";
+
+package grafbase.graphql;
+
+message CompositeSchemaEntity {
+  optional string entity = 1;
+  optional string relation_field_name = 2;
+  optional string key_field_name = 3;
+}
+
+extend google.protobuf.MessageOptions {
+  optional string object_directives = 58301;
+  optional string input_object_directives = 58302;
+}
+
+extend google.protobuf.FieldOptions {
+  optional string output_field_directives = 58301;
+  optional string input_field_directives = 58302;
+  optional CompositeSchemaEntity composite_schemas_entity = 58303;
+}
+
+extend google.protobuf.EnumOptions {
+  optional string enum_directives = 58301;
+}
+
+extend google.protobuf.EnumValueOptions {
+  optional string enum_value_directives = 58301;
+}
+
+extend google.protobuf.ServiceOptions {
+  optional bool default_to_query_fields = 58301;
+  optional bool default_to_mutation_fields = 58302;
+  optional string subgraph_name = 58303;
+  optional string schema_directives = 58304;
+}
+
+extend google.protobuf.MethodOptions {
+  optional string directives = 58301;
+  optional bool is_query = 58302;
+  optional bool is_mutation = 58303;
+}
+
+//!file:composite_schemas_entity.proto
+syntax = "proto3";
+
+import "grafbase/options.proto";
+
+package test.composite;
+
+service ProductService {
+  rpc GetProduct(GetProductRequest) returns (Product);
+  rpc CreateProduct(CreateProductRequest) returns (Product);
+  rpc ListProducts(ListProductsRequest) returns (ListProductsResponse);
+}
+
+// Test messages with composite schema entities
+message Product {
+  string id = 1;
+  string name = 2;
+  string description = 3;
+  // Basic usage - entity User with field id
+  string user_id = 4 [(grafbase.graphql.composite_schemas_entity) = {entity: "User"}];
+  // With custom relation field name
+  string owner_id = 5 [(grafbase.graphql.composite_schemas_entity) = {entity: "User", relation_field_name: "owner"}];
```

Comment:
why not just `grafbase.graphql.derive`? and `relation_field_name` as just `name` maybe